sandbox/cgroup/freezer.go

```diff
@@ -178,10 +190,52 @@ func writeExistingFile(where string, data []byte, mode os.FileMode) error {
 	return errC
 }
 
+var freezeOneV2 = func(ctx context.Context, dir string) error {
+	groupName := filepath.Base(dir)
+	fname := filepath.Join(dir, "cgroup.freeze")
+	if err := writeExistingFile(fname, []byte("1")); err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			//  the group may be gone already
+			return nil
+		}
+		return fmt.Errorf("cannot freeze processes in group %q: %w", groupName, err)
+	}
+
+	// Add an upper bound to the timeout if cgroup is stuck at freeze state
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, maxFreezeTimeout)
+	defer cancel()
+	ticker := time.NewTicker(100 * time.Millisecond)
```

Comment:
Please add `defer ticker.Stop()` with a note that it can be removed once we move to Go 1.23

Reply:
Nice catch, Thanks!